src/xdp-utils.c

```diff
@@ -187,3 +225,26 @@ xdp_connection_track_name_owners (GDBusConnection *connection)
                                       name_owner_changed,
                                       NULL, NULL);
 }
+
+void
+xdp_register_request (Request *request)
+{
+  AppIdData *data;
+
+  G_LOCK (app_ids);
+  data = (AppIdData *)g_hash_table_lookup (app_ids, request->sender);
```

Comment:
There is a race here, where the lookup could fail if the sender died and we got removed from the hashtable.